serde_v8/error.rs

```diff
@@ -1,54 +1,54 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 use std::fmt::Display;
-use std::fmt::{self};
-
-use serde::de;
-use serde::ser;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, thiserror::Error)]
 #[non_exhaustive]
 pub enum Error {
+  #[error("{0}")]
   Message(String),
 
+  #[error("serde_v8 error: invalid type, expected: boolean")]
   ExpectedBoolean,
+  #[error("serde_v8 error: invalid type, expected: integer")]
   ExpectedInteger,
+  #[error("serde_v8 error: invalid type, expected: number")]
   ExpectedNumber,
+  #[error("serde_v8 error: invalid type, expected: string")]
   ExpectedString,
+  #[error("serde_v8 error: invalid type, expected: array")]
   ExpectedArray,
+  #[error("serde_v8 error: invalid type, expected: map")]
   ExpectedMap,
+  #[error("serde_v8 error: invalid type, expected: enum")]
   ExpectedEnum,
+  #[error("serde_v8 error: invalid type, expected: object")]
   ExpectedObject,
+  #[error("serde_v8 error: invalid type, expected: buffer")]
   ExpectedBuffer,
+  #[error("serde_v8 error: invalid type, expected: detachable")]
   ExpectedDetachable,
+  #[error("serde_v8 error: invalid type, expected: external")]
   ExpectedExternal,
 
+  #[error("serde_v8 error: invalid type, expected: utf8")]
   ExpectedUtf8,
+  #[error("serde_v8 error: invalid type, expected: latin1")]
```

Comment:
Can you make these errors show what was actual type?

Reply:
Let me try~

Reply:
This will be handled by @levex as part of https://github.com/denoland/deno/issues/17406